x-pack/plugins/security_solution/public/common/components/event_details/event_details.tsx

```diff
@@ -22,82 +20,87 @@ export enum EventsViewType {
-  margin: 20px 0;
-`;
-
-CollapseLink.displayName = 'CollapseLink';
-
 interface Props {
   browserFields: BrowserFields;
-  columnHeaders: ColumnHeaderOptions[];
   data: TimelineEventsDetailsItem[];
   id: string;
   view: EventsViewType;
-  onUpdateColumns: OnUpdateColumns;
   onViewSelected: (selected: EventsViewType) => void;
   timelineId: string;
-  toggleColumn: (column: ColumnHeaderOptions) => void;
 }
 
-const Details = styled.div`
-  user-select: none;
-`;
+const StyledEuiTabbedContent = styled(EuiTabbedContent)`
+  display: flex;
+  flex: 1;
+  flex-direction: column;
+  overflow: hidden;
 
-Details.displayName = 'Details';
+  > [role='tabpanel'] {
+    display: flex;
+    flex: 1;
+    flex-direction: column;
+    overflow: hidden;
+  }
+`;
 
-export const EventDetails = React.memo<Props>(
-  ({
-    browserFields,
-    columnHeaders,
-    data,
-    id,
-    view,
-    onUpdateColumns,
+const EventDetailsComponent: React.FC<Props> = ({
+  browserFields,
+  data,
+  id,
+  view,
+  onViewSelected,
+  timelineId,
+}) => {
+  const handleTabClick = useCallback((e) => onViewSelected(e.id as EventsViewType), [
     onViewSelected,
-    timelineId,
-    toggleColumn,
-  }) => {
-    const handleTabClick = useCallback((e) => onViewSelected(e.id as EventsViewType), [
-      onViewSelected,
-    ]);
+  ]);
 
-    const tabs: EuiTabbedContentTab[] = useMemo(
-      () => [
-        {
-          id: EventsViewType.tableView,
-          name: i18n.TABLE,
-          content: (
+  const tabs: EuiTabbedContentTab[] = useMemo(
+    () => [
+      {
+        id: EventsViewType.tableView,
+        name: i18n.TABLE,
+        content: (
+          <>
+            <EuiSpacer size="l" />
             <EventFieldsBrowser
               browserFields={browserFields}
-              columnHeaders={columnHeaders}
               data={data}
               eventId={id}
-              onUpdateColumns={onUpdateColumns}
               timelineId={timelineId}
-              toggleColumn={toggleColumn}
             />
-          ),
-        },
-        {
-          id: EventsViewType.jsonView,
-          name: i18n.JSON_VIEW,
-          content: <JsonView data={data} />,
-        },
-      ],
-      [browserFields, columnHeaders, data, id, onUpdateColumns, timelineId, toggleColumn]
-    );
+          </>
+        ),
+      },
+      {
+        id: EventsViewType.jsonView,
+        name: i18n.JSON_VIEW,
+        content: (
+          <>
+            <EuiSpacer size="m" />
+            <JsonView data={data} />
+          </>
+        ),
+      },
+    ],
+    [browserFields, data, id, timelineId]
+  );
 
-    return (
-      <Details data-test-subj="eventDetails">
-        <EuiTabbedContent
-          tabs={tabs}
-          selectedTab={view === 'table-view' ? tabs[0] : tabs[1]}
-          onTabClick={handleTabClick}
-        />
-      </Details>
-    );
-  }
-);
+  const selectedTab = useMemo(() => (view === EventsViewType.tableView ? tabs[0] : tabs[1]), [
```

Comment:
nit -> I do not think that the useMemo around selectedTab is helping anything since tabs is already memoized. I will expect tabs[0] has the same reference if tabs did not change because of the useMemo for tabs